### PR TITLE
Fixed matching patterns used for mocking

### PR DESCRIPTION
**Pull request by ssbarnea**

### CODE REVIEW

**Comment by felixfontein** [comment]
There's another one in https://github.com/ansible-community/ansible-lint/blob/ee28104a47b9672579682693beae12464f338e82/src/ansiblelint/_prerun.py#L229

File: src/ansiblelint/_prerun.py [hunk 1]
```diff
@@ -226,9 +226,14 @@ def _prepare_ansible_paths() -> None:
 
 def _make_module_stub(module_name: str) -> None:
     # a.b.c is treated a collection
-    if re.match(r"\w+\.\w+\.\w+", module_name):
-        namespace, collection, module_file = module_name.split(".")
+    if re.match(r"\w+\.\w+\.[\.\w]+$", module_name):
```
Comment (felixfontein) [inline src/ansiblelint/_prerun.py:229]: Shouldn't this be
```suggestion
    if re.match(r"\w+\.\w+(\.\w+)+$", module_name):
```
if you want to match proper names?

File: .ansible-lint [hunk 2]
```diff
@@ -8,8 +8,9 @@ exclude_paths:
 # Mock modules or roles in order to pass ansible-playbook --syntax-check
 mock_modules:
   - zuul_return
-  - fake_namespace.fake_collection.fake_module
   # note the foo.bar is invalid as being neither a module or a collection
+  - fake_namespace.fake_collection.fake_module
+  - fake_namespace.fake_collection.fake_module.fake_submodule
```
Comment (felixfontein) [inline .ansible-lint:13]:
```suggestion
  - fake_namespace.fake_collection.fake_module
```
I would really avoid advertising that you can have further dots in module names ;-)

File: src/ansiblelint/_prerun.py [hunk 3]
```diff
@@ -226,25 +226,28 @@ def _prepare_ansible_paths() -> None:
 
 def _make_module_stub(module_name: str) -> None:
     # a.b.c is treated a collection
-    if re.match(r"\w+\.\w+\.\w+", module_name):
-        namespace, collection, module_file = module_name.split(".")
-        path = f"{ options.project_dir }/.cache/collections/ansible_collections/{ namespace }/{ collection }/plugins/modules"
+    if re.match(r"^(\w+|\w+\.\w+\.[\.\w]+)$", module_name):
```
Comment (felixfontein) [inline src/ansiblelint/_prerun.py:229]:
```suggestion
    if re.match(r"^(\w+|\w+\.\w+(\.\w+)+)$", module_name):
```

Comment (ssbarnea) [reply src/ansiblelint/_prerun.py:229]: Based on my tests done on https://regex101.com/ mine example worked as intended but the one with nested matches failed.

File: .ansible-lint (hunk 2 above)
Comment (ssbarnea) [reply .ansible-lint:13]: Is not so much about advertising desired but listing it here assures that we do not choke when we encounter it (there it at least one test making use of it).

File: src/ansiblelint/_prerun.py (hunk 3 above)
Comment (felixfontein) [reply src/ansiblelint/_prerun.py:229]: What exactly did fail? I tried several variants, and they all worked.

File: .ansible-lint (hunk 2 above)
Comment (felixfontein) [reply .ansible-lint:13]: In that case, at least use correct terms: `fake_namespace.fake_collection.fake_directory.fake_module`

**Pull request merged**